services/cd-service/pkg/repository/transformer.go

```diff
@@ -1050,6 +1051,11 @@ func (c *ReleaseTrain) Transform(ctx context.Context, state *State) (string, err
 		if envConfig.Upstream == nil {
 			return fmt.Sprintf("Environment %q does not have upstream configured - exiting.", envName), nil
 		}
+		err := state.checkUserPermissions(ctx, envName, "*", auth.PermissionDeployReleaseTrain, c.RBACConfig)
+		if err != nil {
+			return "", err
+		}
+
```

Comment:
here (down below) when calling the deploy transformer, we need to pass along the authentication.

This actually needs to be done, in all cases when a transformer calls another transformer

Reply:
Added the authentication check for `CreateApplicationVersion` as well. Added a test for it.